data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/Pipeline.java

```diff
@@ -255,9 +255,16 @@ private void shutdownExecutorService(final ExecutorService executorService, fina
     List<Future<Void>> publishToSinks(final Collection<Record> records) {
         final int sinksSize = sinks.size();
         final List<Future<Void>> sinkFutures = new ArrayList<>(sinksSize);
-        router.route(records, sinks, (sink, events) ->
-                sinkFutures.add(sinkExecutorService.submit(() -> sink.output(events), null))
+        final List<DataFlowComponent<Sink>> firstSink = sinks.subList(0,1);
+        final List<DataFlowComponent<Sink>> remainingSinks = sinks.subList(1, sinksSize);
+        router.route(records, firstSink, (sink, events) ->
+                sinkFutures.add(sinkExecutorService.submit(() -> sink.output(events, false), null))
         );
+        if (remainingSinks.size() > 0) {
+            router.route(records, remainingSinks, (sink, events) ->
```

Comment:
This is calling the router twice which will require re-evaluating routes when using conditional routing. Thus, conditional routing sinks may have a performance hit.